Придумать задание для изучения исключений не просто.
Предлагаю просто создать свой класс исключений, пронаследовав его от непроверяемого RunTime(это плохо, я знаю).
В конструктор класса передать целочисленную переменную, которая будет кодом нашей ошибки.
Конструктору суперкласса передать сообщение об исключении нашего класса.
Переопределить getMessage и ToString чтобы указывался наш код.
Ошибку с кодом генерировать статическим методом generator().
В классе Main два блока try-catch: во внутреннем генерируем MyException, получаем сообщение от getMessage и пробрасываем во внешний, который обработает все, т.к. является суперклассом и для RunTime и для MyException.
